skimage/metrics/_contingency_table.py

```diff
@@ -27,14 +28,12 @@ def contingency_table(im_true, im_test, *, ignore_labels=(), normalize=False):
         labeled `i` in `im_true` and `j` in `im_test`.
     """
 
-    im_test_r = im_test.ravel()
-    im_true_r = im_true.ravel()
-    ignored = np.zeros(im_true_r.shape, np.bool)
-    for label in ignore_labels:
-        ignored[im_true_r == label] = True
-    data = np.ones(im_true_r.shape)
-    data[ignored] = 0
+    if ignore_labels is None:
+        ignore_labels = []
+    im_test_r = im_test.reshape(-1)
```

Comment:
just curious to know why you changed `im.ravel()` to `im.reshape(-1)`?

Reply:
(it's just for my own curiosity, no problem with the proposal)

Reply:
@emmanuelle The answer is in the .ravel docstring: .reshape is less likely to result in a copy. Which surprised me but after several reminders @rfezzani has finally hammered it into my brain.

Reply:
now that you say it I'm almost sure I already asked @rfezzani this same question. Thanks for answering, we're doing group learning here :-).